### ISSUE TITLE
NullPointerException while exploring new chunks

### ISSUE THREAD

**Issue by MatthiasMann** [CorgiTaco/BYG#301]
**Before filling out this issue: **
**1. For major bugs, we only support BYG's latest version on a per Minecraft version basis(i.e latest for 1.15 is 1.0 while latest for 1.16 is 1.1).**
**2. Please test BYG on its own to confirm whether or not this is a mod conflict**

__**Mod conflict? If so, please clarify which mod is conflicting, if not skip this.**__
I was using the same set of mods before with Biomes O Plenty. I removed BOP, added BYG and created a new world. If it is related to one of the other mods it did not happen before - so I'm not sure if it's a conflict.

__**API/Modloader**__
forge-1.16.5-36.0.15
byg-1.1.10

__**Describe the bug**__
I had the crash 2 times (the first time I created a new world because I didn't liked some of the biomes I discovered - and disabled these in the config by removing them from byg-biomes.json).
Both crashes have the same back trace and happened while I was using a boat going out into the ocean.
The 2nd crash was on a minecraft:river surrounded by minecraft:dessert but from terrain it looked like the river was widening (going into an ocean?)

__**To Reproduce**__
I'm not 100% sure which biome is causing this - but I'm sure it's related to oceans. I can provide coordinates and a world save if required.

**Expected behavior**
No crash

**Screenshots/Videos**
If applicable, add screenshots/video to help explain your problem.

**Additional context**

**Logs**
Use: https://paste.ee/ and paste the links. if log is huge please upload those
Please provide:
Crashlog - https://gist.github.com/MatthiasMann/c8d0417074af587c01e94fd13a6a8665
Debug.log - https://gist.github.com/MatthiasMann/f4e9e54834c17cf2aaf9c1d9cb650aee
Latest.log - https://gist.github.com/MatthiasMann/5c861127d3d7aecbe59fffcf80c821e6
(these are the logs from the 2nd crash)
**Heap Dumps**
*If performance, please provide a heap dump/some sort of profile of your game(I recommend: https://visualvm.github.io/download.html)*
Heap Dump/Profile -

**Comment by MatthiasMann** [comment]
I was able to reproduce the crash with only the following mods:
byg-1.1.10.jar
curios-forge-1.16.5-4.0.4.0.jar  (required by eidolon)
eidolon-0.2.7.jar
Xaeros_Minimap_21.3.0.2_Forge_1.16.5.jar
XaerosWorldMap_1.12.0.2_Forge_1.16.5.jar

on seed -1918365373911821839 flying in creative at x=-275.02, y=141.45, z=-832.07

using my modified byg-biomes.json: https://gist.github.com/MatthiasMann/7fefbcff3b9955ecd14fedf5a6fa6f44

**Comment by CorgiTaco** [comment]
Nothing in the logs points to BYG. Have you found the culprit since?

**Comment by MatthiasMann** [comment]
No, I have since then stopped using both BYG and Eidolon.

**Comment by Elfinlocks** [comment]
> No, I have since then stopped using both BYG and Eidolon.

Did you ever have any luck with this? Removed BOP from my pack and encountering this on brand new worlds all of a sudden.

**Comment by CorgiTaco** [comment]
@Elfinlocks send your logs

**Comment by Elfinlocks** [comment]
> @Elfinlocks send your logs
[latest.log](https://github.com/AOCAWOL/BYG/files/8307301/latest.log)

 Game crashes with a stack overflow error upon entering the Nether in a brand new world. Removing BYG prevents this from happening.

If you'd like to see the specific instance, this is reproducable with a 100% success rate in this release of my modpack. 

https://www.curseforge.com/minecraft/modpacks/astral-engineering/files/3695996

**Comment by Elfinlocks** [comment]
UPDATE: I fixed this by deleting my BYG config

**Comment by CorgiTaco** [comment]
@Elfinlocks i think your issue is that you have a sub biome(ie hills) entry with a weight of 0 in your config, this is a bug on our end i have yet to fix

**Comment by Aleon4201** [comment]
> @Elfinlocks i think your issue is that you have a sub biome(ie hills) entry with a weight of 0 in your config, this is a bug on our end i have yet to fix

I don't think so.
I occurred same crash now, remove byg configs, fix it, see configs, and found its problem.
Here's my byg-nether-biomes.json.
[byg-nether-biomes.json](https://paste-bin.xyz/54969)
This file has other mods' biomes too. (In my case, "theabyss" = The Abyss: Legacy.)
MatthiasMann, Elfinlocks, and I had installed BOP and removed it. Our configs should have BOP's biomes' ID too.
Btw, my crash report is [here](https://paste-bin.xyz/54967).

If you have already found it and been fixing, I'm verry sorry.
But I'm glad if this issue will be fixed on older MC versions too (i.e. 1.16.5).
